box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java

```diff
@@ -0,0 +1,274 @@
+    }
+
+    /**
+     * Helper method for helping with transformation of BoxBatchResponse to UI Model
+     * @param responses the batch response to transform
+     * @return the transformed model
+     */
+    private static InviteCollaboratorsDataWrapper handleCollaboratorsInvited(BoxResponseBatch responses) {
+        int strCode = R.string.box_sharesdk_generic_error; //default generic error
+        boolean mInvitationFailed;
+        String subMssg;
+
+        int alreadyAddedCount = 0;
+        boolean didRequestFail = false;
+        String name = "";
+        HashSet<Integer> failureCodes = generateFailureCodes();
+        List<String> failedCollaboratorsList = new ArrayList<>();
+
+
+        for (BoxResponse<BoxCollaboration> r : responses.getResponses()) {
+            if (!r.isSuccess()) {
+                if (checkIfKnownFailure(r, failureCodes)) {
+                    name = updateFailureStats(r, failedCollaboratorsList);
+                    alreadyAddedCount += name != null ? 1 : 0;
+                }
+                didRequestFail = true;
+            }
+        }
+
+        if (didRequestFail) {
+            String[] result = processAddCollabsRequestFailure(failedCollaboratorsList, name, alreadyAddedCount);
+            strCode = Integer.parseInt(result[0]);
+            subMssg = result[1];
+        } else {
+            String[] result = processAddCollabsRequestSuccess(responses);
+            strCode = Integer.parseInt(result[0]);
+            subMssg = result[1];
+        }
+
+        mInvitationFailed = (didRequestFail && !failedCollaboratorsList.isEmpty());
+
+        return new InviteCollaboratorsDataWrapper(subMssg, strCode, mInvitationFailed);
+    }
+
+    /**
+     * A helper method for updating attributes used for keeping track of stats for failure
+     * @param r the response to get request infos from
+     * @param failedCollaboratorsList list of failed collabs to be updated
+     * @return the name of collaborator that is already added
+     */
+    @VisibleForTesting
+    static String updateFailureStats(BoxResponse<BoxCollaboration> r, List<String> failedCollaboratorsList) {
+        String code = ((BoxException) r.getException()).getAsBoxError().getCode();
+        BoxUser user = (BoxUser) ((BoxRequestsShare.AddCollaboration) r.getRequest()).getAccessibleBy();
+
+        if (alreadyAddedFailure(code)) {
+            return user == null ? "" : user.getLogin();
+        } else {
+            failedCollaboratorsList.add(user == null ? "" : user.getLogin());
+            return null;
+        }
+    }
+
+    /**
+     * Helper method for processing request if all requests were successful
+     * @param responses the responses that was successful
+     * @return index 0 is string resource code, index 1 is the string formatted part of the message.
+     */
+    @VisibleForTesting
+    static String[] processAddCollabsRequestSuccess(BoxResponseBatch responses) {
```

Comment:
Similar comment as above. Rename as a getMethod - so the name tells you what the function returns. This makes it readable in the calling code.

Reply:
I will make the change.